content/docs/getting-started/futures.md

```diff
@@ -183,13 +179,12 @@ fn poll(&mut self) -> Result<Async<usize>, T::Error> {
 
 # Returning `NotReady`
 
-The last section handwaved a bit and said that when a task returns `NotReady`,
-once it transitioned to the ready state, the executor is notifed. This enables
-the executor to be efficient in scheduling tasks.
+The last section handwaved a bit and said that once a Future transitioned to the
+ready state, the executor is notifed. This enables the executor to be efficient
+in scheduling tasks.
 
-When a function returns `Async::NotReady`, it is critical that the executor is
-notified when the state transitions to "ready". Otherwise, the task will hang
-infinitely, never getting run again.
+It is critical that the executor is notified when the state transitions to "ready".
```

Comment:
It is only critical that the executor is notified if the function previously returned `NotReady`.

Perhaps:

> When a function returns `Async::NotReady`, it signals that it is currently not in a ready state and is unable to complete the operation. It is critical that the executor is notified when the state transitions to "ready". Otherwise, the task will hang infinitely, never getting run again.

Reply:
This sounds better to me. I like your suggestion!